components/ml/org.wso2.carbon.ml.core/src/main/java/org/wso2/carbon/ml/core/impl/MLModelHandler.java

```diff
@@ -649,6 +650,14 @@ private void persistModel(long modelId, String modelName, MLModel model) throws
             String storageType = storage.getType();
             String storageLocation = storage.getLocation();
 
+            // if this is a deeplearning model, need to set the storage location for writing
+            // then the sparkdeeplearning model will use ObjectTreeBinarySerializer to write it to the given directory
+            // the DeeplearningModel will be saved as a .bin file
+            if (model.getAlgorithmClass().equalsIgnoreCase(MLConstants.DEEPLEARNING)) {
```

Comment:
to avoid NPEs use; MLConstants.DEEPLEARNING.equalsIgnoreCase(model.getAlgorithmClass())